android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonsProvider.kt

```diff
@@ -10,19 +10,34 @@ package mozilla.components.feature.addons
 interface AddonsProvider {
 
     /**
-     * Provides a list of all available add-ons.
+     * Provides a list of all featured add-ons.
      *
-     * @param allowCache whether or not the result may be provided
-     * from a previously cached response, defaults to true.
-     * @param readTimeoutInSeconds optional timeout in seconds to use when fetching
-     * available add-ons from a remote endpoint.
+     * @param allowCache whether or not the result may be provided from a previously cached response,
+     * defaults to true.
+     * @param readTimeoutInSeconds optional timeout in seconds to use when fetching featured add-ons
+     * from a remote endpoint.
      * @param language indicates in which language the translatable fields should be in, if no
      * matching language is found then a fallback translation is returned using the default language.
      * When it is null all translations available will be returned.
      */
-    suspend fun getAvailableAddons(
+    suspend fun getFeaturedAddons(
         allowCache: Boolean = true,
         readTimeoutInSeconds: Long? = null,
         language: String? = null,
     ): List<Addon>
+
+    /**
+     * Provides a list of [Addon] based on the list of GUIDs passed to it.
+     *
+     * @param readTimeoutInSeconds optional timeout in seconds to use when fetching the add-ons from
+     * a remote endpoint.
```

Comment:
Maybe  we could omit the `remote endpoint` part so the interface will be agnostic from the source, and the implementation could indicate further details from where is fetching :)